meilisearch/index.py

```diff
@@ -512,7 +518,14 @@ def add_documents_raw(
         MeiliSearchApiError
             An error containing details about why Meilisearch can't process your request. Meilisearch error codes are described here: https://docs.meilisearch.com/errors/#meilisearch-errors
         """
-        url = self._build_url(primary_key)
```

Comment:
Why not add this behavior to the `_build_url`? Because this if/else is repeated twice.